aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -160,165 +177,147 @@ module post_mint_reveal_nft::minting {
     /// The source certificate id not found in the signer's account.
     const ETOKEN_ID_NOT_FOUND: u64 = 15;
     /// Can only exchange after the reveal starts.
-    const ECANNOT_EXCHANGE_BEFORE_REVEAL_SRARTS: u64 = 16;
+    const ECANNOT_EXCHANGE_BEFORE_REVEAL_STARTS: u64 = 16;
     /// Can only add unique token uris.
     const EDUPLICATE_TOKEN_URI: u64 = 17;
+    /// The whitelist stage provided is not valid.
+    const EINVALID_STAGE: u64 = 18;
 
     /// Initialize NFTMintConfig for this module.
-    fun init_module(post_mint_reveal_nft_resource_account: &signer) {
-        let resource_signer_cap = resource_account::retrieve_resource_account_cap(post_mint_reveal_nft_resource_account, @source_addr);
-        move_to(post_mint_reveal_nft_resource_account, NFTMintConfig {
+    fun init_module(admin: &signer) {
+        let seed_vec = bcs::to_bytes(&signer::address_of(admin));
```

Comment:
This is not necessary. create_resource_account already does this.

Reply:
I think the code here is trying to construct a resource address that's hard to guess - I think `create_resource_account` only takes in a given seed vector, they don't provide one that makes the resource address hard to guess.

Reply:
create_resource_account takes a signer and adds the corresponding address to the seed by default